packages/core/src/internal/utils/global.ts

```diff
@@ -50,26 +41,41 @@ export function setupCDSGlobal() {
   if (isBrowser()) {
     initializeCDSGlobal();
     setRunningVersion();
+    intializeCDSStateProxy();
   }
 }
 
-function getVersion() {
-  if (isBrowser()) {
-    const log: CDSLog = {
-      versions: window.CDS._version,
-      environment: window.CDS.environment,
-      userAgent: navigator.userAgent,
-      supports: window.CDS._supports,
-      angularVersion: getAngularVersion(false),
-      angularJSVersion: getAngularJSVersion(false),
-      reactVersion: getReactVersion(false),
-      vueVersion: getVueVersion(false),
-      loadedElements: window.CDS._loadedElements,
-    };
-    return log;
+function intializeCDSStateProxy() {
+  if (!window.CDS._stateProxy) {
+    window.CDS._stateProxy = true;
+    window.CDS._state = new Proxy(window.CDS._state, {
+      set: (target: any, key: string, value) => {
+        const detail = { key, prev: (window.CDS._state as any)[key], current: value };
+        target[key] = value;
+        document.dispatchEvent(new CustomEvent('CDS_STATE_UPDATE', { detail }));
+        return true;
+      },
+    });
```

Comment:
todo: add test here and explainer in comment

The proxy wraps the state object to allow us to trigger events when any of the state object properties are updated. This allows services to know when the state was updated even if there are duplicates or multiple versions. The i18n and icon services leverage this to update/rerender components when new values are added.

Reply:
added